p2p/src/swarm/peerdb.rs

```diff
@@ -122,25 +155,64 @@ impl<T: NetworkingService> PeerDb<T> {
         false // TODO: implement
     }
 
-    /// Discover new peer addresses
-    pub fn discover_peers(&mut self, peers: &[types::AddrInfo<T>]) {
-        log::info!("discovered {} new peers", peers.len());
+    /// Check if the peers is part of our active swarm
+    pub fn is_active_peer(&self, peer_id: &T::PeerId) -> bool {
+        std::matches!(self.peers.get(peer_id), Some(Peer::Active(_)))
+    }
 
-        for info in peers.iter() {
-            // TODO: update peer stats
-            if self.peers.contains_key(&info.id) {
-                continue;
-            }
+    /// Get socket address of the next best peer (TODO: in terms of peer score)
+    // TODO: rewrite all of this
+    pub fn best_peer_addr(&mut self) -> crate::Result<Option<T::Address>> {
```

Comment:
Shouldn't this function be called "take_..."? Usually we use "take" or "pop" for things that erase from the container and return the result.